starlette/middleware/__init__.py

```diff
@@ -1,8 +1,10 @@
 import typing
 
+from starlette.types import MiddlewareType
+
 
 class Middleware:
-    def __init__(self, cls: type, **options: typing.Any) -> None:
+    def __init__(self, cls: typing.Type[MiddlewareType], **options: typing.Any) -> None:
```

Comment:
Before this gets merged, maybe using another name as `MiddlewareType` is more suitable? (Perhaps `MiddlewareT`). Having `typing.Type[MiddlewareType]` could lead to some confusion (type of a type? 🤨). Originally, I've named it `MiddlewareType` to indicate that this is a type hint of a middleware hence my suggestion of using `MiddlewareT` instead, the `T` being pretty common to indicate type hints

Reply:
I prefer `MiddlewareType`